google-cloud-storage/lib/google/cloud/storage/policy/bindings.rb

```diff
@@ -0,0 +1,166 @@
+        #                           description: "description of condition",
+        #                           expression: "expr1"
+        #                         }
+        #                       })
+        #   end
+        #
+        class Bindings
+          include Enumerable
+
+          ##
+          # @private Creates a Bindings object.
+          def initialize
+            @bindings = []
+          end
+
+          ##
+          # Adds a binding or bindings to the collection.
+          #
+          # @param [Google::Cloud::Storage::Policy::Binding] bindings One or
+          #   more bindings to be added to the policy owning the collection.
+          #
+          # @example
+          #   require "google/cloud/storage"
+          #
+          #   storage = Google::Cloud::Storage.new
+          #   bucket = storage.bucket "my-todo-app"
+          #
+          #   bucket.policy requested_policy_version: 3 do |p|
+          #     p.bindings.insert({
+          #                         role: "roles/storage.admin",
+          #                         members: ["user:owner@example.com"],
+          #                         condition: {
+          #                           title: "test-condition",
+          #                           description: "description of condition",
+          #                           expression: "expr1"
+          #                         }
+          #                       })
+          #   end
+          #
+          def insert *bindings
+            bindings = coerce_bindings(*bindings)
+            @bindings += bindings
+          end
+
+          ##
+          # Deletes a binding or bindings from the collection.
+          #
+          # @param [Google::Cloud::Storage::Policy::Binding] bindings One or
+          #   more bindings to be removed from the policy owning the
+          #   collection.
+          #
+          # @example
+          #   require "google/cloud/storage"
+          #
+          #   storage = Google::Cloud::Storage.new
+          #   bucket = storage.bucket "my-todo-app"
+          #
+          #   bucket.policy requested_policy_version: 3 do |p|
+          #     p.bindings.remove({
+          #                         role: "roles/storage.admin",
+          #                         members: ["user:owner@example.com"],
+          #                         condition: {
+          #                           title: "test-condition",
+          #                           description: "description of condition",
+          #                           expression: "expr1"
+          #                         }
+          #                       })
+          #   end
+          #
+          def remove *bindings
+            bindings = coerce_bindings(*bindings)
+            @bindings -= bindings
+          end
+
+          ##
+          # Calls the block once for each binding in the collection, passing
+          # the binding object as parameter.
```

Comment:
Maybe clarify that this always yields `Binding` objects even if you insert a Hash.

Reply:
Fixed.